src/renderer/sw_engine/tvgSwFill.cpp

```diff
@@ -214,7 +223,12 @@ bool _prepareLinear(SwFill* fill, const LinearGradient* linear, const Matrix* tr
     fill->linear.dy = y2 - y1;
     auto len = fill->linear.dx * fill->linear.dx + fill->linear.dy * fill->linear.dy;
 
-    if (len < FLOAT_EPSILON) return true;
+    if (len < FLOAT_EPSILON) {
+        if (tvg::zero(fill->linear.dx) && tvg::zero(fill->linear.dy)) {
+            fill->forceSolid = true;
+        }
+        return true;
```

Comment:
```
return false;
```

Reply:
there is no reason to return false, this is a valid case. right now if I return false nothing will be rendered